commit 9caec4bf1d0126fa5f2fcd21852958bccd2a4c18
Author: Paolo Bonzini <pbonzini@redhat.com>
Date:   Mon Dec 14 07:59:15 2020 -0500

    KVM: x86: remove bogus #GP injection
    
    There is no need to inject a #GP from kvm_mtrr_set_msr, kvm_emulate_wrmsr will
    handle it.
    
    Reviewed-by: Tom Lendacky <thomas.lendacky@amd.com>
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

diff --git a/arch/x86/kvm/mtrr.c b/arch/x86/kvm/mtrr.c
index 7f0059aa30e1..f472fdb6ae7e 100644
--- a/arch/x86/kvm/mtrr.c
+++ b/arch/x86/kvm/mtrr.c
@@ -84,12 +84,8 @@ bool kvm_mtrr_valid(struct kvm_vcpu *vcpu, u32 msr, u64 data)
 	} else
 		/* MTRR mask */
 		mask |= 0x7ff;
-	if (data & mask) {
-		kvm_inject_gp(vcpu, 0);
-		return false;
-	}
 
-	return true;
+	return (data & mask) == 0;
 }
 EXPORT_SYMBOL_GPL(kvm_mtrr_valid);